internal/common/resource.go

```diff
@@ -290,7 +290,7 @@ func (r *reconcileBuilder) createOrUpdateWithImmutableSpec(obj client.Object, f
 			return OperationResultNone, err
 		}
 		if err := mutate(f, key, obj); err != nil {
-			return OperationResultNone, err
+			return OperationResultNone, fmt.Errorf("error creating object %s/%s: %w", obj.GetNamespace(), obj.GetName(), err)
```

Comment:
This change is a leftover from my testing. It doesn't need to be in this PR. We can add it in a later PR.